### ISSUE TITLE
1.17 - Map Editor Crash

### ISSUE THREAD

**Issue by doofus-01** [wesnoth/wesnoth#7954]
### Game and System Information

- **Version:**
- 1.17.21+dev 
- **Downloaded from:**
- here
- **Build info:**
- c9182b952b8
- **OS:**
Linux

### Description of the bug

Open a map in the editor, Wesnoth quits when you scroll around.
```
Caught general 'St12out_of_range' exception:
vector::_M_range_check: __n (which is 0) >= this->size() (which is 0)
```
I tried a few mainline MP and campaign maps, it did not always happen, but with SoF map `5_Hills_of_the_Shorbear_Clan.map` it seems consistent, though you might need to scroll around a bit.

It happens in both Full-screen and Windowed.

### Steps to reproduce the behavior

See above

**Comment by Pentarctagon** [comment]
This sounds like it should be fixed by #7951?

**Comment by doofus-01** [comment]
It does appear to fix it, should be merge before release for sure.
